### ISSUE TITLE
When creating a multi-host data source, TargetSessionAttributes cannot be specified

### ISSUE THREAD

**Issue by tfreyx** [npgsql/doc#263]

**Comment by roji** [comment]
Yep, should update the docs in https://www.npgsql.org/doc/failover-and-load-balancing.html to show data source samples.

Coming to think of it, we could add an overload of BuildMultiHost which accepts the target session attributes directly...

**Comment by roji** [comment]
You should only be seeing that exception when calling BuildMultiHost() on NpgsqlDataSourceBuilder, not with a simple UseNpgsql() call as above - can you please double-check, and if you're still seeing the error, post a minimal repro?

**Comment by fmendez89** [comment]
> You should only be seeing that exception when calling BuildMultiHost() on NpgsqlDataSourceBuilder, not with a simple UseNpgsql() call as above - can you please double-check, and if you're still seeing the error, post a minimal repro?

Ok, I found the problem, but I don't know if it's a problem or unsupported configuration.

```c#
c.UseNpgsql("Host=localhost,localhost:6432;Database=test;Username=test;Password=test;Load Balance Hosts=true;Target Session Attributes=prefer-standby",
        config =>
        {
            **config.ConfigureDataSource(dsopt =>
            {
                dsopt.EnableParameterLogging();
            });**
            
        });
```

If I use ConfigureDataSource to, for example, enable parameter logging, it throws the exception.

**Comment by roji** [comment]
@fmendez89 thanks, that all makes sense to me now. When you call ConfigureDataSource() (you can remove EnableParameterLogging()), EFCore.PG creates an NpgsqlDataSourceBuilder for you internally, and that doesn't allow specifying `Target Session Attributes` via the connection string as the error says.

I've opened https://github.com/npgsql/efcore.pg/issues/3495 to explore a fix for this.

What you'll need to do here, is what's specified in the [docs](https://www.npgsql.org/doc/failover-and-load-balancing.html?tabs=7): create an NpgsqlMultiHostDataSource yourself (outside of EF), get a wrapper for the required Target Session, and then pass that to EF's UseNpgsql().

```c#
var dataSourceBuilder = new NpgsqlDataSourceBuilder("Host=server1,server2;Username=test;Password=test");
await using var dataSource = dataSourceBuilder.BuildMultiHost();
var preferStandbyDataSource = dataSource.WithTargetSession(TargetSessionAttributes.PreferStandby);

// Pass preferStandbyDataSource to EF's UseNpgsql()
```

In other words, ConfigureDataSource() currently is unsupported with `Target Session Attribute`.

/cc @NinoFloris @vonzshik

**Comment by fmendez89** [comment]
> [@fmendez89](https://github.com/fmendez89) thanks, that all makes sense to me now. When you call ConfigureDataSource() (you can remove EnableParameterLogging()), EFCore.PG creates an NpgsqlDataSourceBuilder for you internally, and that doesn't allow specifying `Target Session Attributes` via the connection string as the error says.
> 
> I've opened [npgsql/efcore.pg#3495](https://github.com/npgsql/efcore.pg/issues/3495) to explore a fix for this.
> 
> What you'll need to do here, is what's specified in the [docs](https://www.npgsql.org/doc/failover-and-load-balancing.html?tabs=7): create an NpgsqlMultiHostDataSource yourself (outside of EF), get a wrapper for the required Target Session, and then pass that to EF's UseNpgsql().
> 
> var dataSourceBuilder = new NpgsqlDataSourceBuilder("Host=server1,server2;Username=test;Password=test");
> await using var dataSource = dataSourceBuilder.BuildMultiHost();
> var preferStandbyDataSource = dataSource.WithTargetSession(TargetSessionAttributes.PreferStandby);
> 
> // Pass preferStandbyDataSource to EF's UseNpgsql()
> In other words, ConfigureDataSource() currently is unsupported with `Target Session Attribute`.
> 
> /cc [@NinoFloris](https://github.com/NinoFloris) [@vonzshik](https://github.com/vonzshik)

Thanks for the reply @roji Very helpful.
